### PR TITLE
Added LB diagram and some more detail

### PR DESCRIPTION
**Pull request by dgquintas**
Fixes https://github.com/grpc/grpc.github.io/issues/232

### CODE REVIEW

File: doc/load-balancing.md [hunk 1]
```diff
@@ -53,14 +53,67 @@ unavailability or health issues. The load balancer will make any necessary
...
+
+1. On startup, the gRPC client issues a name resolution request for the service.
+   The name will resolve to one or more IP addresses to gRPC servers, a hint on
+   whether the IP address(es) point to a load balancer or not, and also return a
+   client config.
+2. The gRPC client connects to a gRPC Server.
+   1. If the name resolution has hinted that the endpoint is a load balancer,
+      the client will attempt to open a stream to the load balancer service. The
+      server may respond in only one of the following ways.
+      1. `status::UNIMPLEMENTED`. There is no loadbalancing in use. The client
```
Comment (zhangkun83) [inline doc/load-balancing.md]: In the revised security design, the naming system's response will indicate whether the endpoint is loadbalancer or not. I don't think we need this fallback behavior any more. I think it's appropriate to fail in this case.

Comment (dgquintas) [inline doc/load-balancing.md]: Done. PTAL.

**Comment by dgquintas** [comment]
I've made some small tweaks from @iamqizhao's input. PTAL.

**Comment by iamqizhao** [comment]
LGTM

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* One or more commits are not associated with a GitHub user. The email used in the Git commit must match a verified email address in a GitHub user profile. For more information, please see https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/

_Regards,
CLA GitHub bot_

**Pull request merged**